custom_components/example_sensor/sensor.py

```diff
@@ -21,28 +25,17 @@ def setup_platform(
 class ExampleSensor(SensorEntity):
     """Representation of a Sensor."""
 
-    def __init__(self):
-        """Initialize the sensor."""
-        self._state = None
-
-    @property
-    def name(self) -> str:
-        """Return the name of the sensor."""
-        return 'Example Temperature'
+    _attr_name = "Example Temperature"
+    _attr_native_unit_of_measurement = TEMP_CELSIUS
+    _attr_device_class = SensorDeviceClass.TEMPERATURE
+    _attr_state_class = SensorStateClass.MEASUREMENT
 
-    @property
-    def state(self):
-        """Return the state of the sensor."""
-        return self._state
-
-    @property
-    def unit_of_measurement(self) -> str:
-        """Return the unit of measurement."""
-        return TEMP_CELSIUS
+    def __init__(self) -> None:
+        """Initialize the sensor."""
```

Comment:
```suggestion
```
The init is not used, and can be removed

Reply:
I was thinking to put something there just to maintain it as it's an example but could not figure out something good...

Reply:
Ok. I went with it and removed __init__